Readme.md

```diff
@@ -0,0 +1,46 @@
+# Setup Steps
+
+- If using docker see [Docker](#docker)
+
+- If python virtual environment is already created for your local docs directory see [Usage](#usage) steps else start with [Installation](#installation).
+
+- Checkout the docs directory and execute the below commands.
+
+## Docker
+
+```console
+> docker build . -t sphinx:latest
+> docker run --rm -it -h sphinx -v $(pwd):/mnt/host/docs -w /mnt/host/docs -p 8000:8000 sphinx
+> sphinx-autobuild ./source/ ./build/html/ --port 8000 --host 0.0.0.0
+```
+
+## Installation
+
+```console
+> python3 -m venv venv
+> source venv/bin/activate
+> pip3 install -r requirements.txt
+```
+
+## Usage
+
+```console
+> source venv/bin/activate
+> make html -b spelling -b linkcheck
+> sphinx-autobuild ./source/ ./build/html/ --port 8000 --host 0.0.0.0
```

Comment:
How does the web server get installed to host the content on your local machine? Do we need to add those steps under Installation?

Reply:
We should make the installation of and building for a web server optional (I'm assuming that's what this command is doing). The user can also view the entire built directory structure as a local file by using a "file://" format URL on his browser.

Reply:
of course, you can view it using file://.
But, if you want to live preview it while editing the files at the same time. Then this is needed.